bin/cylc-get-host-metrics

```diff
@@ -0,0 +1,214 @@
+          or more valid mount directory PATHs (as listed under 'Mounted on')
+          within the filesystem of localhost. Multiple PATH options can be
+          specified via a comma-delimited list, each becoming a key under the
+          top-level disk space key.
+
+   If no options are specified, --load and --memory are invoked by default."""
+
+
+import sys
+if '--use-ssh' in sys.argv[1:]:
+    sys.argv.remove('--use-ssh')
+    from cylc.remote import remrun
+    if remrun():
+        sys.exit(0)
+
+import os
+import re
+from subprocess import Popen, PIPE, CalledProcessError
+import json
+
+import cylc.flags
+from cylc.option_parsers import CylcOptionParser as COP
+
+
+"""Templates for extracting desired metric data via regular expression.
+
+   Example command outputs, with '*' enclosing data to extract with templates:
+
+   LOAD_TEMPLATES
+     $ uptime
+     17:52:58 up  8:08,  2 users,  load average: *0.13*, *0.09*, *0.07*
+
+   MEMORY_TEMPLATES
+     $ free -k
+                  total       used       free     shared    buffers     cached
+     Mem:       6116852    5017920    1098932    54964     130328    1837404
+     -/+ buffers/cache:    3050188  *3066664*
+     Swap:      4190204      14624    4175580
+
+   DISK_TEMPLATES, e.g. for mount directory path option '--disk-space=/boot'
+     $ df -Pk /boot
+     Filesystem            1024-blocks      Used Available Capacity Mounted on
+     /dev/md0                    96978     33904   *57856*      37% /boot."""
+LOAD_TEMPLATES = ['uptime', (1, 5, 15), re.compile(
+                  r'load average:\s(\d+\.?\d+),\s(\d+\.?\d+),\s(\d+\.?\d+)')]
+MEMORY_TEMPLATES = ['free -k', re.compile(
+                    r'\-/\+ buffers/cache:\s+\d+\s+(\d+)')]
+DISK_TEMPLATES = ['df -Pk', re.compile(
+                  r'Mounted on\n\S+\s+\d+\s+\d+\s+(\d+)\s+')]
+
+
+def main():
+    parser = COP(__doc__, comms=True, noforce=True, argdoc=[])
+
+    parser.add_option(
+        "--load", "-l",
+        help="1, 5 and 15 minute load averages from the 'uptime' command",
+        action="store_true", default=False, dest="load")
+
+    parser.add_option(
+        "--memory", "-m",
+        help="Total free RAM memory, in KB, from the 'free -k' command.",
+        action="store_true", default=False, dest="memory")
+
+    parser.add_option(
+        "--disk-space",
+        help="Available disk space, in KB, from the 'df -Pk' command",
+        action="store", default=None, dest="disk")
+
+    options, args = parser.parse_args()
+    print get_host_metric(options.load, options.memory, options.disk)
+
+    sys.exit()
+
+
+def run_command_string(command_str):
+    """Return the result from running a command string in a local shell.
+
+       Return stdout if command successful, else raise CalledProcessError."""
+    # Inputs to process are internal only, so 'shell=True' is safe here.
+    process = Popen(command_str, shell=True, stdin=open(os.devnull),
```

Comment:
Do we actually need to use `shell=True`?

Reply:
When I remove it, only the load data via the ``uptime`` command seems to come through (i.e. only the ``--load`` tests pass). So I assume we do? I'll have to have a more detailed look at quite why this is.